pkg/vgmanager/vgmanager_controller.go

```diff
@@ -0,0 +1,206 @@
+	r.Log = log.FromContext(ctx).WithName(ControllerName)
+	r.Log.Info("reconciling", "lvmcluster", req)
+	res, err := r.reconcile(ctx, req)
+	if err != nil {
+		r.Log.Error(err, "reconcile error")
+	}
+	r.Log.V(1).Info("reconcile complete", "result", res)
+	return res, err
+
+}
+func (r *VGReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	lvmCluster := &lvmv1alpha1.LVMCluster{}
+	err := r.Client.Get(ctx, req.NamespacedName, lvmCluster)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	r.Log.V(1).Info("listing block devices")
+	//  list block devices
+	blockDevices, badRows, err := internal.ListBlockDevices()
+	if err != nil {
+		msg := fmt.Sprintf("failed to list block devices: %v", err)
+		r.Log.Error(err, msg, "lsblk.BadRows", badRows)
+		return ctrl.Result{}, err
+	} else if len(badRows) > 0 {
+		r.Log.Error(err, "could not parse all the lsblk rows", "lsblk.BadRows", badRows)
+	}
+
+	// load lvmd config
+	lvmdConfig := &lvmdCMD.Config{
+		SocketName: topolvm.DefaultLVMdSocket,
+	}
+
+	cfgBytes, err := os.ReadFile(ConfigFilePath)
+	if os.IsNotExist(err) {
+		r.Log.Info("lvmd config file doesn't exist, will create")
+		err = yaml.Unmarshal(cfgBytes, &lvmdConfig)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	} else if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	oldLVMDConfig := *lvmdConfig
+
+	// avoid having to iterate through device classes multiple times, map from name to config index
+	deviceClassMap := make(map[string]int)
+	for i, deviceClass := range lvmdConfig.DeviceClasses {
+		deviceClassMap[deviceClass.Name] = i
+	}
+
+	// filter out block devices
+	remainingValidDevices, delayedDevices, err := r.filterAvailableDevices(blockDevices)
+	if err != nil {
+		_ = err
+	}
+
+	for _, deviceClass := range lvmCluster.Spec.DeviceClasses {
+		// ignore deviceClasses whose LabelSelector doesn't match this node
+		// NodeSelectorTerms.MatchExpressions are ORed
+		node := &corev1.Node{}
+		selectsNode, err := NodeSelectorMatchesNodeLabels(node, deviceClass.NodeSelector)
+		if err != nil {
+			r.Log.Error(err, "failed to match nodeSelector to node labels")
+			continue
+		}
+		if !(selectsNode && ToleratesTaints(deviceClass.Tolerations, node.Spec.Taints)) {
+			continue
+		}
+		_, found := deviceClassMap[deviceClass.Name]
+		if !found {
+			lvmdConfig.DeviceClasses = append(lvmdConfig.DeviceClasses, &lvmd.DeviceClass{
+				Name:        deviceClass.Name,
+				VolumeGroup: deviceClass.Name,
+			})
+		}
+		var matchingDevices []internal.BlockDevice
+		remainingValidDevices, matchingDevices, err = filterMatchingDevices(remainingValidDevices, deviceClass)
+		if err != nil {
+			r.Log.Error(err, "could not filterMatchingDevices")
+			continue
+		}
+		if len(matchingDevices) > 0 {
+			// create/update VG and update lvmd config
+			err = r.addMatchingDevicesToVG(matchingDevices, deviceClass.Name)
+			if err != nil {
+				r.Log.Error(err, "could not prepare volume groups")
```

Comment:
Include the name of the VG in the message

Reply:
done.